src/app/utils/firebase.js

```diff
@@ -16,171 +16,502 @@ const FireBaseTools = {
+          const dateNumber = startDate.getDate() < 10 ? '0' + (startDate.getDate()) : (startDate.getDate());
+          // Create the key for the new DateObject in the form "YEAR-MONTH-DATE"
+          const newDateObject = startDate.getFullYear() + '-' + monthNumber + '-' + dateNumber;
+          // Add the JSON date key
+          switch (startDate.getDay()) {
+
+            case 0: // Sunday
+              if (givenWeekDay[0] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 1: // Monday
+              if (givenWeekDay[1] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 2: // Tuesday
+              if (givenWeekDay[2] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 3: // Wednesday
+              if (givenWeekDay[3] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 4: // Thursday
+              if (givenWeekDay[4] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 5: // Friday
+              if (givenWeekDay[5] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 6: // Saturday
+              if (givenWeekDay[6] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            default:
+              console.log('There was an error in retrieving the day of the week');
+            //TODO add a proper error check?
+          }
+        }
+
+        // check next date (startDate acts as our iterator in this loop so it takes the value of the next day)
+        const newDate = startDate.setDate(startDate.getDate() + 1);
+        startDate = new Date(newDate);
+      }
+      // return a promise of a timetable object with all the dates of a given course section path
+      return timetable;
+    }).catch(error => ({
+      errorCode: error.code,
+      errorMessage: error.message,
+    }));
+    /* eslint-enable */
+  },
+
+  getSections: (courseName) => {
+    const sections = [];
+    /* eslint-disable */
+    return sectionsRef.child(courseName).once('value').then(function(snap) {
+      snap.forEach(function(childSnap) {
+        sections.push({section:childSnap.key, maxPat:childSnap.child('MaxPat').val(), component:childSnap.child('Component').val()})
+      });
+      return sections;
+    }).catch(error => ({
+      errorCode: error.code,
+      errorMessage: error.message,
+    }));
+    /* eslint-enable */
+  },
+
+   getUserEvents: () => {
```

Comment:
This might not be absolutely necessary right now, but it would be nice if there were more comments in this code for better readability.